third_party/nvfuser/csrc/index_compute.cpp

```diff
@@ -1876,14 +1876,24 @@ std::vector<Val*> Index::getStrides(const TensorView* tv) {
 std::vector<Val*> Index::getRootIndices(
     const TensorView* tv,
     const std::vector<kir::ForLoop*>& loops,
-    const IndexFromIdGraph& index_from_id_graph) {
+    const IndexFromIdGraph& index_from_id_graph,
+    bool from_concrete) {
   auto root_dom = tv->getMaybeRFactorDomain();
-  auto indexing = index_from_id_graph.index;
+  // when from_concrete is true, we can generate root indices separately. For
+  // example, when generate index for  T1[I0, I1], if from_concrete is false,
+  // the result of this method is {0, I0 * S1 + I1}; if from_concrete is true,
+  // the result of this method is {I0 * S1, I1}.
```

Comment:
Instead of completely disabling contiguous merge, could we pass the `override_index` to `getTensorIndexFromIdGraph`, and inside `getTensorIndexFromIdGraph`, we add `override_index` into `index_parameters.initial_concrete_id_index` at here?
https://github.com/csarofeen/pytorch/blob/c6bb6318f2c97b24cdcd9bdfbbf10c73ebc9131f/third_party/nvfuser/csrc/lower_index_compute.cpp#L814
I am not very familiar with this code, need @naoyam to confirm.

Reply:
The benefit of contig indexing is that we can save decomposing an index into the domains that are are merged. Since we need to get to the root domain anyway, contig indexing doesn't seem to be beneficial in this case. 

We disable contig indexing for domains with halo. I think we could do the same thing for indirectly accessed domains as well.

Reply:
@naoyam I don't have further concerns other than this. You can approve and merge if you are OK with this PR.

Reply:
I don't think this is what we want. I'll update the PR.

Reply:
Thanks!

Reply:
And the rest of `scatterOp` is heavyly related to this PR, I will push new code as soon as this PR is merged.

Reply:
None of the Python tests fails either.

@ftxj Can you add a C++ test that fails without this change?

Reply:
The code here will only be called when we need to use the `override_index` to do `lowerDstIndex`. I don't think all the Python/C++ tests need this. The code here will be called by `scatter`, but `scatter` in not in our devel branch. This PR just as a part of `scatter` op. When I add the test to call this code, many other code of `scatter op` will be put into this PR.

Reply:
So, looks like this change is not necessary for this PR. Can you move this change to another PR where it's required?

Reply:
Sure.